gimmemotifs/motif/_comparison.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2009-2021 Simon van Heeringen <simon.vanheeringen@gmail.com>
+#
+# This module is free software. You can redistribute it and/or modify it under
+# the terms of the MIT License, see the file COPYING included with this
+# distribution.
+"""Comparison functions for Motif class"""
+from math import log
+
+import numpy as np
+from scipy.stats import pearsonr
+
+from gimmemotifs.utils import make_equal_length
+
+
+def ic_pos(self, row1, row2=None):
+    """Calculate the information content of one position.
+
+    Returns
+    -------
+    score : float
+        Information content.
+    """
+    if row2 is None:
+        row2 = [0.25, 0.25, 0.25, 0.25]
+
+    score = 0
+    for a, b in zip(row1, row2):
+        if a > 0:
+            score += a * log(a / b) / log(2)
+    return score
+
+
+def other_ic_pos(self, row1, row2, bg=None):
+    if bg is None:
+        bg = [0.25, 0.25, 0.25, 0.25]
+
+    score = 0
+    score_a = 0
+    score_b = 0
+
+    for a, b, pbg in zip(row1, row2, bg):
+        score += abs(a * log(a / pbg) / log(2) - b * log(b / pbg) / log(2))
+        score_a += a * log(a / pbg) / log(2)
+        score_b += b * log(b / pbg) / log(2)
+
+    return (score_a + score_b) / 2 - score
+
+
+def pcc(self, ppm1, ppm2, pos):
+    """ """
+    # xxCATGYT
+    # GGCTTGYx
+    # pos = -2
+    ppm1, ppm2 = make_equal_length(ppm1, ppm2, pos, truncate="both")
+
+    # Compute pearson correlation between aligned parts of the motif
+    r = np.array([pearsonr(x, y)[0] for x, y in zip(ppm1, ppm2)])
+    r[np.isnan(r)] = 0
+
+    return r.sum()
+
+
+def ic(self, ppm1, ppm2, pos, bg=None, bg_factor=1):
```

Comment:
Should some of these functions be private?

I'm thinking of `ic`, `ppc`, `ic_pos`, `other_ic_pos`, and the like?